### PR TITLE
Clean up client-go left overs after converting to runtime library.

### PR DESCRIPTION
**Pull request by awels**
Clean unused functions from controller utils.go
Turn util_test.go into gingko test.
Moved functions from util to proper controller if only used by that controller.

Signed-off-by: Alexander Wels <awels@redhat.com>

**What this PR does / why we need it**:
Clean up most remnants of the client-go objects in the CDI controller code.
Cleanup pkg/controller/util.go and remove unused functions and structs.
Switch util_test.go to use gingko framework.
Explicitly did not change leader election mechanism because of known bug in the version of the library we are using.
**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/controller/config-controller.go [hunk 1]
```diff
@@ -63,6 +61,7 @@ func (r *CDIConfigReconciler) Reconcile(req reconcile.Request) (reconcile.Result
 		log.Error(err, "Unable to create CDIConfig")
 		return reconcile.Result{}, err
 	}
+	log.Info("CDIConfig", "config", config)
```
Comment (maya-r) [inline pkg/controller/config-controller.go:64]: Leftover debug output?

File: pkg/controller/import-controller_test.go [hunk 2]
```diff
@@ -409,11 +409,152 @@ var _ = Describe("Import test env", func() {
...
+		table.Entry("return true on endpoint with no port, and cdi configmap exists, and host defined", endpointNoPort, true, host, true),
+		table.Entry("return true on endpoint with port, and cdi configmap exists, and host with port", endpointWithPort, true, hostWithPort, true),
+		table.Entry("return false on endpoint with no port, and cdi configmap exists, and host with port", endpointNoPort, true, hostWithPort, false),
+		table.Entry("return false on endpoint with port, and cdi configmap exists, and host defined", endpointWithPort, true, host, false),
+		table.Entry("return false on endpoint with no port, and no cdi configmap exists, and blank host", endpointNoPort, false, "", false),
+		table.Entry("return false on blank endpoint, and cdi configmap exists, and host defined", "", true, host, false),
+	)
+})
+
+var _ = Describe("getInsecureTLS", func() {
```
Comment (maya-r) [inline pkg/controller/import-controller_test.go:495]: Do you mean getContentType?
(this test doesn't do anything TLS-related)

File: pkg/controller/util_test.go [hunk 3]
```diff
@@ -14,881 +17,149 @@ import (
...
-	}
-
-	type args struct {
-		pvc    *corev1.PersistentVolumeClaim
-		objs   []runtime.Object
-		result string
-	}
-
-	for _, arg := range []args{
-		{&corev1.PersistentVolumeClaim{}, nil, ""},
```
Comment (maya-r) [inline pkg/controller/util_test.go:653]: In the interest of completeness, it looks like this case was dropped in the new getCertConfigMap tests (2 cases vs. 3).

**Comment by maya-r** [comment]
otherwise lgtm

File: pkg/controller/config-controller.go (hunk 1 above)
Comment (awels) [reply pkg/controller/config-controller.go:64]: yes thanks for spotting that.

File: pkg/controller/import-controller_test.go (hunk 2 above)
Comment (awels) [reply pkg/controller/import-controller_test.go:495]: I did mean that, copy and paste strikes again.

File: pkg/controller/util_test.go (hunk 3 above)
Comment (awels) [reply pkg/controller/util_test.go:653]: You are right, its not a very interesting case, but we should keep it for completeness.

**Comment by awels** [comment]
/hold
need to fix a bunch of tests

File: pkg/controller/config-controller.go [hunk 4]
```diff
@@ -36,9 +34,9 @@ import (
 
 // CDIConfigReconciler members
 type CDIConfigReconciler struct {
-	Client                 client.Client
-	CdiClient              cdiclientset.Interface
-	K8sClient              kubernetes.Interface
+	Client client.Client
```
Comment (mhenriks) [inline pkg/controller/config-controller.go:37]: Do these fields need to be exported?

Comment (awels) [reply pkg/controller/config-controller.go]: Nope, good point, fixed.

**Comment by awels** [comment]
Looks like some GH or PROW issue, it was merged, so closing.

**Pull request merged**